platform/ios/src/MGLMapboxEvents.m

```diff
@@ -665,11 +656,13 @@ - (void)locationManager:(MGLLocationManager *)locationManager didUpdateLocations
         double accuracy = 10000000;
         double lat = floor(loc.coordinate.latitude * accuracy) / accuracy;
         double lng = floor(loc.coordinate.longitude * accuracy) / accuracy;
+        double horizontalAccuracy = loc.horizontalAccuracy;
```

Comment:
According to the [`horizontalAccuracy`](https://developer.apple.com/reference/corelocation/cllocation/1423599-horizontalaccuracy) documentation, a value of −1 indicates an invalid `coordinate`. We should check for this case (and also `coordinate` being invalid) and avoid sending the event.